Update Bitbucket pipeline (#1491)
semgrep/semgrep-docs: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx

image: atlassian/default-image:latest

pipelines:
  branches:

>     main:

Reviewer: ```suggestion
    # Change to your default branch if different from main
    main:
```

  branches:
    main:
    - step:
        image: semgrep/semgrep
        name: Semgrep Scan on Push
        script:
          - semgrep ci

  pull-requests:
    '**': # This applies to pull requests for all branches
      - step:
          name: 'Run Semgrep scan with current branch'
          deployment: dev # https://support.atlassian.com/bitbucket-cloud/docs/set-up-and-monitor-deployments/
          image: semgrep/semgrep
          name: Semgrep Scan on PR
          script:
            # The following variables are required to set up a scan connected to Semgrep Cloud Platform:

>             - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN

Reviewer: I don't think we ought to remove this line :) Can you restore it?

Reviewer: Yeah, good point - we may need to not only restore this but make sure it's in all the right places.

Author: @s-santillan SEMGREP_APP_TOKEN should still be in there. That is just my mistake.

Reviewer 2: Added it into the 3 events. Resolving.

>             - export SEMGREP_BASELINE_REF="origin/main"

Reviewer: ```suggestion
            # Change to your default branch if different from main
            - export SEMGREP_BASELINE_REF="origin/main"
```

>   # additinal setup for cron in BitBucket https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule

Reviewer: ```suggestion
  # Trigger job manually. For cron in Bitbucket, see: https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule
```

Reviewer: Based on our convo I understood this automatically gives a manual trigger, but if that's not right we can adjust the comment accordingly.
